### PR TITLE
[TEST] Adds mutate function for all metric aggregation tests

### PR DESCRIPTION
**Pull request by colings86**
Relates to #25929

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/InternalDateHistogram.java [hunk 1]
```diff
@@ -277,6 +277,31 @@ public String getWriteableName() {
...
+    /**
+     * Getter for testing only.
+     */
+    long getMinDocCount() {
+        return minDocCount;
+    }
+
+    /**
+     * Getter for testing only.
+     */
```
Comment (javanna) [inline core/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/InternalDateHistogram.java]: I wonder if we need such javadoc comments. It's pretty self-explanatory that they are there for tests, those are the only classes that call them.

File: core/src/main/java/org/elasticsearch/search/aggregations/metrics/cardinality/HyperLogLogPlusPlus.java [hunk 2]
```diff
@@ -446,7 +446,7 @@ public int hashCode(long bucket) {
     public boolean equals(long bucket, HyperLogLogPlusPlus other) {
         return Objects.equals(p, other.p) &&
                 Objects.equals(algorithm.get(bucket), other.algorithm.get(bucket)) &&
-                Objects.equals(getComparableData(bucket), getComparableData(bucket));
+                Objects.equals(getComparableData(bucket), other.getComparableData(bucket));
```
Comment (javanna) [inline core/src/main/java/org/elasticsearch/search/aggregations/metrics/cardinality/HyperLogLogPlusPlus.java]: good catch :)

File: core/src/test/java/org/elasticsearch/search/aggregations/metrics/geobounds/InternalGeoBoundsTests.java [hunk 3]
```diff
@@ -102,4 +103,54 @@ protected void assertFromXContent(InternalGeoBounds aggregation, ParsedAggregati
...
+            negLeft += between(1, 20);
+            break;
+        case 6:
+            negRight += between(1, 20);
+            break;
+        case 7:
+            wrapLongitude = wrapLongitude == false;
+            break;
+        case 8:
+        default:
```
Comment (javanna) [inline core/src/test/java/org/elasticsearch/search/aggregations/metrics/geobounds/InternalGeoBoundsTests.java]: out of curiosity: why default is also the last case? shouldn't we rather be strict and throw exception if the randomization goes out of the supported bounds?

**Review by javanna — APPROVED** [review]
left a few minors, LGTM though

File: core/src/test/java/org/elasticsearch/search/aggregations/metrics/geobounds/InternalGeoBoundsTests.java (hunk 3 above)
Comment (colings86) [inline core/src/test/java/org/elasticsearch/search/aggregations/metrics/geobounds/InternalGeoBoundsTests.java]: Hmmm maybe we should throw here actually. My thinking had been that the default case would not be reached anyway so it didn't matter where it goes but actually I see that if the bounds of the random number are wrong we should probably throw an exception so we can correct the test. I'll make this change in all the test classes in this PR and https://github.com/elastic/elasticsearch/pull/26058

Comment (javanna) [inline core/src/test/java/org/elasticsearch/search/aggregations/metrics/geobounds/InternalGeoBoundsTests.java]: thanks!

**Pull request merged**